lib/src/command/add.dart

```diff
@@ -170,42 +237,73 @@ class AddCommand extends PubCommand {
   /// If a version constraint is provided when the `--path` or any of the
   /// `--git-<option>` options are used, a [PackageParseError] will be thrown.
   ///
-  /// If both `--path` and any of the `--git-<option>` options are defined,
-  /// a [PackageParseError] will be thrown.
+  /// Packages must either be a git, hosted, sdk, or path package. Mixing of
+  /// options is not allowed and will cause a [PackageParseError] to be thrown.
   ///
   /// If any of the other git options are defined when `--git-url` is not
   /// defined, an error will be thrown.
   Pair<PackageRange, dynamic> _parsePackage(String package) {
     ArgumentError.checkNotNull(package, 'package');
 
-    if ([path != null, hasGitOptions, hasHostOptions, sdk != null]
-            .where((element) => element == true)
-            .length >=
-        2) {
-      usageException(
-          'Packages must either be a git, hosted, sdk, or path package.');
+    final _conflictingFlagSets = [
+      ['git-url', 'git-ref', 'git-path'],
+      ['hosted-url'],
+      ['path'],
+      ['sdk'],
+    ];
+
+    for (final flag
+        in _conflictingFlagSets.expand((s) => s).where(argResults.wasParsed)) {
+      final conflictingFlag = _conflictingFlagSets
+          .where((s) => !s.contains(flag))
+          .expand((s) => s)
+          .firstWhere(argResults.wasParsed, orElse: () => null);
+      if (conflictingFlag != null) {
+        usageException(
+            'Packages can only have one source, pub add flags "--$flag" and '
```

Comment:
```suggestion
            'Packages can only have one source, `pub add` flags "--$flag" and '
```

In general, I think we tend to use ` around CLI commands and flags in messages. We should perhaps look at what other parts of pub is doing, just to be sure we are consistent.

Reply:
I took a look at the other parts of pub, and it seems to be a tie between double-quotes and backticks for CLI commands, and no quotes for flags, unless it seems to be part of a larger command. 

**CLI commands**

Double quotes (non-exhaustive):
1. https://github.com/dart-lang/pub/blob/master/lib/src/entrypoint.dart#L530
1. https://github.com/dart-lang/pub/blob/master/lib/src/command/deps.dart#L252
1. https://github.com/dart-lang/pub/blob/master/lib/src/command/downgrade.dart#L18

Backticks (non-exhaustive):
1. https://github.com/dart-lang/pub/blob/master/lib/src/command/outdated.dart#L535-L548
1. https://github.com/dart-lang/pub/blob/master/lib/src/sdk/flutter.dart#L27
1. https://github.com/dart-lang/pub/blob/master/lib/src/solver/report.dart#L155

**Flags**

No quotes:
1. https://github.com/dart-lang/pub/blob/master/lib/src/global_packages.dart#L632
1. https://github.com/dart-lang/pub/blob/master/lib/src/command/downgrade.dart#L43
1. https://github.com/dart-lang/pub/blob/master/lib/src/command/get.dart#L43
1. https://github.com/dart-lang/pub/blob/master/lib/src/command/global_activate.dart#L57

Backtick:
1. https://github.com/dart-lang/pub/blob/master/lib/src/validator/null_safety_mixed_mode.dart#L48

I could create a separate PR to standardize all the CLI commands and flags, just let me know!